app/models/donation.rb

```diff
@@ -1,36 +1,37 @@
 require 'pagarme'
 
 class Donation < ActiveRecord::Base
+  store_accessor :customer
   belongs_to :widget
   has_one :mobilization, through: :widget
   has_one :organization, through: :mobilization
 
-  after_create :capture_transaction
+  after_create :create_transaction
   after_create :send_mail
 
-  def find_transaction
-    PagarMe::Transaction.find_by_id(self.token)
+  def new_transaction
+    PagarMe::Transaction.new({
+      :card_hash => self.card_hash,
+      :amount => self.amount,
+      :payment_method => self.payment_method,
+      :split_rules => split_rules,
+      :metadata => {
+        :widget_id => self.widget.id,
+        :mobilization_id => self.mobilization.id,
+        :organization_id => self.organization.id,
+        :city => self.organization.city,
+        :email => self.customer["email"] }
+    })
   end
 
-  def capture_transaction
+  def create_transaction
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Assignment Branch Condition size for create_transaction is too high. [16.4/15]](https://www.codacy.com/app/jlcfigueira/hub-api/file/2528509798/issues/source?bid=3141163&fileBranchId=3282765#l27)